substrate/frame/revive/src/exec.rs

```diff
@@ -1378,6 +1382,10 @@ where
 	}
 
 	fn delegate_call(&mut self, code_hash: H256, input_data: Vec<u8>) -> Result<(), ExecError> {
+		// We reset the return data now, so it is cleared out even if no new frame was executed.
+		// This is for example the case for unknown code hashes or creating the frame fails.
+		*self.last_frame_output_mut() = Default::default();
```

Comment:
isn't it what we do here already

https://github.com/paritytech/polkadot-sdk/blob/4f0f71a1468bedc734bfb69ac92cfbcf2960e686/substrate/frame/revive/src/exec.rs#L993-L1003

need to double check the code, but maybe this is not needed overthere with these changes

Reply:
also nit maybe a clear_output would look nicer

Reply:
The code in `fn run()` guarantees that the last saved output is empty upon running the new frame. If we want to make the caller responsible for this I can remove the code in `fn run()`.

Reply:
right we need both